### PR TITLE
DDF 1413

### PR DESCRIPTION
**Pull request by kguerriero**
[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/122)

### CODE REVIEW

**Comment by kguerriero** [comment]
@wmcnalli  Will hero

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java [hunk 1]
```diff
@@ -479,20 +481,44 @@ public DeleteResponse delete(DeleteRequest deleteRequest) throws IngestException
...
         }
 
-        /* 1. Query first for the records */
+        if (identifiers.size() <= DEFAULT_PAGING_SIZE) {
+            deleteListOfMetacards(deletedMetacards, identifiers, attributeName);
+        } else {
+            List<? extends Serializable> identifierPaged = null;
+            int currPagingSize = 0;
+
+            for (currPagingSize = DEFAULT_PAGING_SIZE; currPagingSize < identifiers.size(); currPagingSize += DEFAULT_PAGING_SIZE) {
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java]: This is fine, but you might consider partitioning the list with http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Lists.html#partition(java.util.List,%20int) and iterating that list of lists.

**Comment by coyotesqrl** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 539 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-JwcriHFM3z5_GtWRC3-)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L539)):</sup>
If this method is only being used for deletes, it should probably be named appropriately.

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClient.java, line 72 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-Jwcrr5FgMNpJ9MSdkfN)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClient.java#L72)):</sup>
Might be worth finding a common place to set this constant. Is it something that we want to be configurable?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/122)

**Comment by kcwire** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 90 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-JwhjP2mxXDrLR8dfupD)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L90)):</sup>
This really isn't a paging size. It's actually the maximum configured Boolean clauses per the solrconfig.xml. I would update this constant to share the name and value of the config property.
https://github.com/codice/ddf/blob/master/catalog/solr/catalog-solr-external-provider/src/main/resources/solr/conf/solrcatalogconfig.xml#L430

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 487 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-JwhjnlowkJxMrfVHGRF)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L487)):</sup>
identifiersPaged?

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 508 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-Jwhm8OcTFsc7pmrkYOW)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L508)):</sup>
If you don't need the QueryResponse you could just make your getQueryResponseMethod return the SolrDocumentList.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/122)

**Comment by kguerriero** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 487 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-JwhjnlowkJxMrfVHGRF)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L487)):</sup>
This is the subset of identifiers that are used in the construction of the boolean clause. I'll change the name of the constant (DEFAULT_PAGING_SIZE) so this will need to be changed as well. 

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 539 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-JwcriHFM3z5_GtWRC3-)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L539)):</sup>
I modified this method to return a SolrDocumentList so I'll change the name to match this. Currently deleting is not being handled by this method. 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/122)

**Comment by kguerriero** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClient.java, line 0 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-Jwcrr5FgMNpJ9MSdkfN)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClient.java#L0)):</sup>
I could be wrong but I can't think of a reoccurring situation where we would want to change the value of this. 

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/122)

**Comment by kcwire** [comment]
Review status: 2 of 3 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 90 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/122#-JwwpmJ_V3zK3TsHCx7U)** ([raw file](https://github.com/codice/ddf/blob/b28f21762bb328a579dd3e0a2722e38fe2122077/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L90)):</sup>
I would make this match the value of the config.xml file (1024). And add a comment of where this value comes from

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java, line 487 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/122#-JwhjnlowkJxMrfVHGRF)** ([raw file](https://github.com/codice/ddf/blob/14d2d530659e7d3d9321e383c45de7d75d841ea4/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java#L487)):</sup>
I merely meant it looks like you left out the "s"

---

<sup>**[catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClient.java, line 72 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/122#-Jwwq7RBksl2zVhu6Qf_)** ([raw file](https://github.com/codice/ddf/blob/b28f21762bb328a579dd3e0a2722e38fe2122077/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClient.java#L72)):</sup>
Should be the same name and value as in the SolrCatalogProvider - which makes me tend to agree with Richard in that this should moved to a common constant.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/122)

**Comment by wmcnalli** [comment]
Reviewed 1 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/122)

**Comment by wmcnalli** [comment]
Reviewed 1 of 3 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/122)

**Comment by wmcnalli** [comment]
:+1:

**Pull request merged**